include/nccl_ofi_api.h

```diff
@@ -6,13 +6,15 @@
 #ifndef NET_OFI_API_H_
 #define NET_OFI_API_H_
 
-#include "nccl-headers/net.h"
-#include "nccl-headers/error.h"
 
-#ifdef _cplusplus
+#ifdef __cplusplus
 extern "C" {
 #endif
 
+#include "nccl-headers/net.h"
+#include "nccl-headers/error.h"
+#include "nccl_ofi_config_bottom.h"
```

Comment:
I had previously put export lines on all of these, then dropped that hunk before pushing, and this should be removed fully.

That being said, what do you propose I do here if I had wanted to keep those export lines? `nccl_ofi_api.h` is an externally facing header and shouldn't require `config.h`, but we've moved all of our macros into this `config_bottom.h` file.

Maybe a better way is to create `nccl_ofi_macros.h` and pull that in config_bottom.h? Not sure.

Reply:
nccl_ofi_api.h isn't external facing; there is no external facing header for the project (because it's really the nccl-headers we import from our consumer).  Every .c/.cc file in the project should include config.h first, so you should never get here.

Reply:
if *every single file* in the project should `#include "config.h"`, is there a good reason to not just have this injected on the compiler command line? I had proposed this previously [here](https://github.com/aws/aws-ofi-nccl/pull/351/commits/9aeb73a8720dd025d154b3068aa55b91712292cb) on a rejected PR and I feel like it simplifies this considerably. Here is yet another example of a header that requires config.h, and only works because every consumer of it luckily had already included it prior to including this file.